pkg/operator/resources.go

```diff
@@ -191,6 +191,12 @@ func generateStatefulSetSpec(
 	d config.Deployment,
 	shard int32,
 ) (*apps_v1.StatefulSetSpec, error) {
+
+	shards := minShards
+	if reqShards := d.Agent.Spec.Prometheus.Shards; reqShards != nil && *reqShards > 1 {
```

Comment:
`d.Agent` could be nil too, right?

Reply:
It shouldn't be, this code only gets invoked we're operating on a non-nil resource hierarchy